app/controllers/katello/api/v2/activation_keys_controller.rb

```diff
@@ -305,13 +306,36 @@ def find_content_view_environments
         find_cve_for_single
       elsif params[:content_view_environments] || params[:content_view_environment_ids]
         @content_view_environments = ::Katello::ContentViewEnvironment.fetch_content_view_environments(
-              labels: params[:content_view_environments],
-              ids: params[:content_view_environment_ids],
-              organization: @organization || @activation_key&.organization)
+          candlepin_names: params[:content_view_environments],
+          ids: params[:content_view_environment_ids],
+          organization: @organization || @activation_key&.organization)
+        if @content_view_environments.blank?
+          handle_errors(candlepin_names: params[:content_view_environments],
+          ids: params[:content_view_environment_ids])
+        end
       end
+      handle_blank_cve_params
       @organization ||= @content_view_environments.first&.organization
     end
 
+    def handle_blank_cve_params
```

Comment:
Should this be moved to mutli cv params handling thingie, so that both hosts and aks can use it . Or are they wildly different between AK and Hosts ?

Reply:
Difference between AKs and hosts here is that AKs can intentionally have no content view environments, but hosts require at least one.

Reply:
Hmm see if you can add a `require_cve: false`  some param like that or `require_cve?` method in the controller and do like `object.try(:require_cve, false)`

Reply:
Actually looks like the logic is quite different. Lets keep it the way it is.